test/routing/route_test.cc

```diff
@@ -726,6 +729,582 @@ TEST_F(RouteLaneSRelationTest, EvaluateComputeLaneSRangeRelationSameLaneSRanges)
+    EXPECT_CALL(id_index_, DoGetLane(kLaneIdB)).WillRepeatedly(Return(&lane_b_));
+    EXPECT_CALL(id_index_, DoGetLane(kLaneIdC)).WillRepeatedly(Return(&lane_c_));
+    EXPECT_CALL(*(road_geometry_ptr_), DoById()).WillRepeatedly(ReturnRef(id_index_));
+    const std::vector<api::RoadPosition> kStartRoadPositionsPhaseZero{
+        api::RoadPosition{&lane_a_, kStartLanePositionPhase}};
+    const std::vector<api::RoadPosition> kEndRoadPositionsPhaseZero{api::RoadPosition{&lane_a_, kEndLanePositionPhase}};
+    const std::vector<api::RoadPosition> kStartRoadPositionsPhaseOne{
+        api::RoadPosition{&lane_b_, kStartLanePositionPhase}};
+    const std::vector<api::RoadPosition> kEndRoadPositionsPhaseOne{api::RoadPosition{&lane_b_, kEndLanePositionPhase}};
+    const std::vector<api::RoadPosition> kStartRoadPositionsPhaseTwo{
+        api::RoadPosition{&lane_c_, kStartLanePositionPhase}};
+    const std::vector<api::RoadPosition> kEndRoadPositionsPhaseTwo{api::RoadPosition{&lane_c_, kEndLanePositionPhase}};
+
+    phase_zero_ = std::make_unique<Phase>(kPhaseZeroIndex, kLaneSRangeTolerance, kStartRoadPositionsPhaseZero,
+                                          kEndRoadPositionsPhaseZero, kLaneSRangesPhaseZero, road_network_.get());
+    phase_one_ = std::make_unique<Phase>(kPhaseOneIndex, kLaneSRangeTolerance, kStartRoadPositionsPhaseOne,
+                                         kEndRoadPositionsPhaseOne, kLaneSRangesPhaseOne, road_network_.get());
+    phase_two_ = std::make_unique<Phase>(kPhaseTwoIndex, kLaneSRangeTolerance, kStartRoadPositionsPhaseTwo,
+                                         kEndRoadPositionsPhaseTwo, kLaneSRangesPhaseTwo, road_network_.get());
+    EXPECT_CALL(*road_geometry_ptr_, do_linear_tolerance()).WillRepeatedly(Return(kLinearTolerance));
+  }
+
+ protected:
+  SegmentMock segment_two_;
+  std::unique_ptr<Phase> phase_two_;
+};
+
+TEST_F(RouteComputeLaneSRouteCaseATest, ComputeLaneSRoute) {
+  const api::RoadPosition kRoadPositionA{&lane_a_, kQueryLanePosition};
+  const api::RoadPosition kRoadPositionB{&lane_b_, kQueryLanePosition};
+  const api::RoadPosition kRoadPositionC{&lane_c_, kQueryLanePosition};
+  const api::LaneSRoute kExpectedLaneSRouteFromA({kLaneSRangeA, kLaneSRangeB, kLaneSRangeC});
+  const api::LaneSRoute kExpectedLaneSRouteFromB({kLaneSRangeB, kLaneSRangeC});
+  const api::LaneSRoute kExpectedLaneSRouteFromC({kLaneSRangeC});
+  const Route dut({*phase_zero_, *phase_one_, *phase_two_}, road_network_.get());
+
+  EXPECT_TRUE(AssertCompare(IsEqual(kExpectedLaneSRouteFromA, dut.ComputeLaneSRoute(kRoadPositionA))));
+  EXPECT_TRUE(AssertCompare(IsEqual(kExpectedLaneSRouteFromB, dut.ComputeLaneSRoute(kRoadPositionB))));
+  EXPECT_TRUE(AssertCompare(IsEqual(kExpectedLaneSRouteFromC, dut.ComputeLaneSRoute(kRoadPositionC))));
+}
+
+// Case B:
+// - Feature under evaluation: can move to the left at the beginning in the same Phase.
```

Comment:
It's confusing to me how this is considered "move to the left" given that, which standing one lane A and facing the route's destination, we need to move to the "right" onto lane B. In what sense is this moving to the left?

Similar question for the other test cases below.

Nice test cases by the way! Thank you for the thoroughness.

Reply:
You are right. I think I wrote these tests from the backwards planning point of view, which is not the forward traversing of the agent.
Fixed in test case descriptions!